src/components/base/Button/index.tsx

```diff
@@ -0,0 +1,22 @@
+import styled from '@emotion/styled';
+import type { HTMLAttributes, ReactElement } from 'react';
+import type { ButtonSizeType, ButtonColorType } from '../../../types';
+import { BUTTON_SIZE } from '../../../utils/constants/size';
+import { THEME_COLOR } from '../../../utils/constants/color';
+
+interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  // Text 컴포넌트의 TextProps import 해오기
+  // children: ReactElement<TextProps>;
+  type: 'button' | 'reset' | 'submit';
+  size: ButtonSizeType;
+  color: ButtonColorType;
```

Comment:
```suggestion
  size?: ButtonSizeType;
  color?: ButtonColorType;
```
size 와 color에 대해서는 optional로 부여가 되어야하는거 아닌가요? default size와 color가 있어서, prop을 전달 받지 못했을때 default 값으로 render되는 형태가 좋을것 같은데 어떠신가요

Reply:
default를 처음에 생각 안 했다가 뒤늦게 디폴트를 주는 걸 생각했는데
디폴트를 줄 거면 말씀하신대로 옵셔널로 주는 게 맞을 것 같아요